src/app/models/body-type-enum.ts

```diff
@@ -0,0 +1,69 @@
+// The types and codes need to be lowercase for the API.
+
+import { MultiOptions } from '@forms/models/options.model';
+
+export enum BodyTypeDescription {
+  ARTICULATED = 'articulated',
+  BOX = 'box',
+  DOUBLE_DECKER = 'double decker',
+  FLAT = 'flat',
+  OTHER = 'other',
+  PETROL_OR_OIL_TANKER = 'petrol/oil tanker',
+  REFUSE = 'refuse',
+  REFRIGERATED = 'refrigerated',
+  SINGLE_DECKER = 'single decker',
+  SKELETAL = 'skeletal',
+  SKIP_LOADER = 'skip loader',
+  TIPPER = 'tipper',
+}
+
+export function getBodyTypesAsOptions(): MultiOptions {
+  return Object
+    .entries(BodyTypeDescription)
+    .map(([, value]) => ({ value, label: value }));
+}
+
+export enum BodyTypeCode {
+  A = 'a',
+  B = 'b',
+  C = 'c',
+  D = 'd',
+  F = 'f',
+  K = 'k',
+  M = 'm',
+  O = 'o',
+  P = 'p',
+  R = 'r',
+  S = 's',
+  T = 't',
+  X = 'x'
+}
+
+export function getCode(bodyType: BodyTypeDescription): BodyTypeCode {
```

Comment:
Could you create a hash map instead of this long switch case statement

Reply:
Seconded

Reply:
I knew there was a cleaner way to achieve this but couldn't think of any at the time. Thanks for the great suggestion!